scene/gui/text_edit.cpp

```diff
@@ -7595,7 +7629,16 @@ void TextEdit::_backspace_internal(int p_caret) {
 		}
 
 		int from_line = to_column > 0 ? to_line : to_line - 1;
-		int from_column = to_column > 0 ? (to_column - 1) : (text[to_line - 1].length());
+		int from_column = 0;
+		if (to_column == 0) {
+			from_column = text[to_line - 1].length();
+		} else {
+			if (caret_mid_grapheme_enabled || !backspace_deletes_composite_character_enabled) {
+				from_column = to_column - 1;
+			} else {
+				from_column = TS->shaped_text_prev_character_pos(text.get_line_data(to_line)->get_rid(), to_column);
+			}
```

Comment:
```suggestion
		} else if (caret_mid_grapheme_enabled || !backspace_deletes_composite_character_enabled) {
			from_column = to_column - 1;
		} else {
			from_column = TS->shaped_text_prev_character_pos(text.get_line_data(to_line)->get_rid(), to_column);
```
Collapse for readability, this way you don't have to make sure that there isn't anything in the `else` block that is shared between the two statements here, using the `else if` communicates this

Reply:
There are three other places in the existing code where this readability improvement would apply - should I fix those while I'm in here? See lines 2812, 2869 and 3145.

Reply:
I'd say to leave unrelated code alone to keep the history clean